server/sonar-server/src/test/java/org/sonar/server/computation/monitoring/CEQueueStatusImplConcurrentTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.monitoring;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CEQueueStatusImplConcurrentTest {
+  private ExecutorService executorService = Executors.newFixedThreadPool(10, new ThreadFactory() {
+    private int cnt = 0;
+
+    @Override
+    public Thread newThread(Runnable r) {
+      return new Thread(r, CEQueueStatusImplConcurrentTest.class.getSimpleName() + cnt++);
+    }
+  });
+  private CEQueueStatusImpl underTest = new CEQueueStatusImpl();
+
+  @After
+  public void tearDown() throws Exception {
+    executorService.shutdownNow();
+  }
+
+  @Test
+  public void test_concurrent_modifications_in_any_order() throws InterruptedException {
+    for (Runnable runnable : buildShuffleCallsToUnderTest()) {
+      executorService.submit(runnable);
+    }
+
+    executorService.awaitTermination(5, TimeUnit.SECONDS);
```

Comment:
it was defensive because I don't know how Travis will handle 40 tasks on 10 threads... I'll lower it down and see how it behave in the branch.